lib/chef/provider/git.rb

```diff
@@ -103,8 +105,11 @@ def action_sync
         end
       end
 
-      def git_minor_version
-        @git_minor_version ||= Gem::Version.new( git("--version").stdout.split.last )
+      def git_gem_version
+        output = git("--version").stdout
+        match = GIT_VERSION_PATTERN.match(output)
+        raise ArgumentError, "unparsable git version number #{output}" unless match
```

Comment:
it'd probably make more sense to leave the version `nil` or `false` or something, while outputting a Chef::Log.warn message and then something like:

```ruby
def git_has_single_branch_option?
  !git_gem_version.nil? && git_gem_version >= Gem::Version.new("1.7.10")
end

[...]

  clone_cmd << "--no-single-branch" if new_resource.depth && git_has_single_branch_option?
```

Reply:
I'm of the same opinion as @lamont-granquist. I'd love to see us assume that the `git` available to us will do most of what we need it to do, do a best-effort parsing of the version, and just be selective about the `--no-single-branch` flag.

Reply:
Good suggestions implemented.  Thanks!